### PR TITLE
[GH-#2661] Added missing setBinaryStream function

### LINKED ISSUE #2661: Add missing setBinaryStream in DB API
Add missing `setBinaryInput` for prepared statements in the DB API and update the documentation.

### PR DESCRIPTION
**Pull request by nedelcho-delchev-tues**
### What does this PR do?
Adds missing setBinaryStream function in DB API and updated the [documentation](https://www.dirigible.io/api/database/database/#preparedstatement).

### What issues does this PR fix or reference?
fixes #2661

### CODE REVIEW

File: components/api-database/src/main/resources/META-INF/dirigible/db/database.js [hunk 1]
```diff
@@ -206,6 +206,14 @@ function PreparedStatement(internalStatement) {
 		this.native.setNull(index, sqlType);
 	};
 
+	this.setBinaryStream = function (parameter, inputStream, length) {
+		if (length !== undefined && length !== null) {
```
Comment (vmutafov) [inline components/api-database/src/main/resources/META-INF/dirigible/db/database.js:210]: `if(length)` is enough here

File: components/api-database/src/main/resources/META-INF/dirigible/db/extensions/db.d.ts [hunk 2]
```diff
@@ -335,6 +335,14 @@ declare module "@dirigible/db" {
          */
         setClob(index:number, value);
 
+        /**
+         * Sets a parameter of type binary strem
+         * @param index
+         * @param inputStream
+         * @param length length of the input stream. If not provided gets all the binary data
+         */
+        setBinaryStream(index: number, inputStream, length?)
```
Comment (vmutafov) [inline components/api-database/src/main/resources/META-INF/dirigible/db/extensions/db.d.ts:344]: We should add the type for the `length` parameter

Comment (vmutafov) [reply components/api-database/src/main/resources/META-INF/dirigible/db/extensions/db.d.ts:344]: And do we know the `inputStream`'s type?

**Review by vmutafov — CHANGES_REQUESTED** [review]

**Review by ThuF — APPROVED** [review]
+1 Seems good to me.

**Pull request merged**